app/src/main/java/org/simple/clinic/location/Coordinates.kt

```diff
@@ -9,4 +10,34 @@ data class Coordinates(
 
     @ColumnInfo(name = "longitude")
     val longitude: Double
-)
+) {
+
+  companion object {
+
+    // Earth is not a perfect sphere so this is an approximate value.
+    private const val EARTH_RADIUS = 6371.0088
+
+    /**
+     * Haversine formula determines the great-circle distance between two points
+     * on a sphere given their longitudes and latitudes.
+     *
+     * Source: https://github.com/jasonwinn/haversine
+     * Modified to use with Kotlin.
+     */
+    fun haversineDistance(start: Coordinates, end: Coordinates): Distance {
```

Comment:
Deleting `Haversine` object and moving the calculation to `Coordinates.haversineDistance()` was @vinaysshenoy's feedback here: https://github.com/simpledotorg/simple-android/pull/487#discussion_r256693845

Not important for this PR, but we later realized that using this function directly in `FacilityListItemBuilder` is not a good idea from testability POV so we ended up moving this again to a class called `DistanceCalculator`.

Reply:
Calculating distance should not be a part of `Coordinates`.
Since you will do this as part of the next PR, I will leave the review of this.